File: src/undo/UndoCommand.cpp
#include "undo/UndoCommand.h"

UndoCommand::UndoCommand(const QString &description,
                         std::function<void()> undo,
                         std::function<void()> redo)
    : m_description(description)
    , m_undo(std::move(undo))
    , m_redo(std::move(redo))
{
}

void UndoCommand::undo()
{
    if (m_undo) m_undo();
}

void UndoCommand::redo()
{
    if (m_redo) m_redo();
}

QString UndoCommand::description() const
{
    return m_description;
}

File: csharp/src/Checklist.Core/Undo/UndoCommand.cs
namespace Checklist.Undo;

public class UndoCommand
{
    private readonly Action _undo;
    private readonly Action _redo;

    public UndoCommand(string description, Action undo, Action redo)
    {
        Description = description;
        _undo = undo;
        _redo = redo;
    }

    public string Description { get; }

    public void Undo() => _undo();

    public void Redo() => _redo();
}